Makefile

```diff
@@ -64,9 +64,22 @@ all:
 	# BIOS boot partition must be defined with an absolute offset.  The
 	# particular value here is 2049, or 0x01 0x08 0x00 0x00 in little-endian.
 	/bin/echo -n -e '\x01\x08\x00\x00' | dd of=pc-core.img seek=500 bs=1 conv=notrunc
-	cp $(SNAPCRAFT_STAGE)/usr/lib/shim/shimx64.efi.dualsigned shim.efi.signed
-	cp $(SNAPCRAFT_STAGE)/usr/lib/grub/x86_64-efi-signed/grubx64.efi.signed grubx64.efi
+	# We must pull in dualsigned shim & grub with UC20 signature
+	# Do it by hand, as snapcraft doesn't have support for PPA archives yet
+	# And yet people try to rebuild this gadget snap
```

Comment:
but we do now have ppa support in snapcraft.yaml proper, it is non-experimental and in the stable channel available everywhere

Reply:
example: https://github.com/snapcore/pi-gadget/pull/73

Reply:
Actually we can't use package-repositories here, because we must pin to UC20 signing key binaries, and the version numbers of those in the UC20 ppa and the primary archive can be the same. Or like primar archive be ahead. And there is no pinning support. Thus direct pulls like that are best at the moment.

Reply:
does package-repositories not support pinning to the version you need and specifying to use the version from the PPA over the main archive? if not, sounds like it should support that

Reply:
it does not currently have an option to specify a pin priority.